s2a/src/main/java/io/grpc/s2a/handshaker/ProtoUtil.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.s2a.handshaker;
+
+import com.google.common.collect.ImmutableSet;
+
+/** Converts proto messages to Netty strings. */
+final class ProtoUtil {
+  /**
+   * Converts {@link Ciphersuite} to its {@link String} representation.
+   *
+   * @param ciphersuite the {@link Ciphersuite} to be converted.
+   * @return a {@link String} representing the ciphersuite.
+   * @throws AssertionError if the {@link Ciphersuite} is not one of the supported ciphersuites.
+   */
+  static String convertCiphersuite(Ciphersuite ciphersuite) {
+    switch (ciphersuite) {
+      case CIPHERSUITE_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256:
+        return "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256";
+      case CIPHERSUITE_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384:
+        return "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384";
+      case CIPHERSUITE_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256:
+        return "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256";
+      case CIPHERSUITE_ECDHE_RSA_WITH_AES_128_GCM_SHA256:
+        return "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256";
+      case CIPHERSUITE_ECDHE_RSA_WITH_AES_256_GCM_SHA384:
+        return "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384";
+      case CIPHERSUITE_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256:
+        return "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256";
+      default:
+        throw new AssertionError(
+            String.format("Ciphersuite %d is not supported.", ciphersuite.getNumber()));
+    }
+  }
+
+  /**
+   * Converts a {@link TLSVersion} object to its {@link String} representation.
+   *
+   * @param tlsVersion the {@link TLSVersion} object to be converted.
+   * @return a {@link String} representation of the TLS version.
+   * @throws AssertionError if the {@code tlsVersion} is not one of the supported TLS versions.
+   */
+  static String convertTlsProtocolVersion(TLSVersion tlsVersion) {
+    switch (tlsVersion) {
+      case TLS_VERSION_1_3:
+        return "TLSv1.3";
+      case TLS_VERSION_1_2:
+        return "TLSv1.2";
+      case TLS_VERSION_1_1:
+        return "TLSv1.1";
+      case TLS_VERSION_1_0:
+        return "TLSv1";
+      default:
+        throw new AssertionError(
```

Comment:
Using an error for an unexpected argument is asking for huge trouble. From what I can tell, this is not currently checked by the caller _and_ this comes from a remote system.

Reply:
Yes, the tls version info is coming from S2A. Would it be appropriate if we threw an ```IllegalArgumentException``` here, and dealt with it in the caller (SslContextFactory)?

Reply:
Done in https://github.com/grpc/grpc-java/pull/11588/commits/9fdaa3684369dd6112d887e3dea2893d4573f61f